pkg/poddisruptionbudget/poddisruptionbudget_test.go

```diff
@@ -0,0 +1,374 @@
+
+		builderResult := NewBuilder(testSettings, testCase.testName, testCase.testNamespace)
+
+		if testCase.expectedError != "" {
+			assert.NotNil(t, builderResult)
+		} else {
+			if testCase.apiClientNil {
+				assert.Nil(t, builderResult)
+			} else {
+				assert.NotNil(t, builderResult)
+				assert.Equal(t, testCase.expectedBuilder.Definition.Name, builderResult.Definition.Name)
+				assert.Equal(t, testCase.expectedBuilder.Definition.Namespace, builderResult.Definition.Namespace)
+				assert.Equal(t, testCase.expectedError, builderResult.errorMsg)
+			}
+		}
+	}
+}
+
+func TestPDBPull(t *testing.T) {
+	testCases := []struct {
+		testName            string
+		testNamespace       string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		apiClientNil        bool
+	}{
+		{
+			testName:            "test1",
+			testNamespace:       "testNamespace",
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			apiClientNil:        false,
+		},
+		{
+			testName:            "test2",
+			testNamespace:       "testNamespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "PodDisruptionBudget object test2 does not exist in namespace testNamespace",
+			apiClientNil:        false,
+		},
+		{
+			testName:            "",
+			testNamespace:       "testNamespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "PodDisruptionBudget 'name' cannot be empty",
+			apiClientNil:        false,
+		},
+		{
+			testName:            "test3",
+			testNamespace:       "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "PodDisruptionBudget 'namespace' cannot be empty",
+			apiClientNil:        false,
+		},
+		{
+			testName:            "test4",
+			testNamespace:       "testNamespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "apiClient is nil",
+			apiClientNil:        true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testResourceQuota := generatePDB(testCase.testName, testCase.testNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testResourceQuota)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		if testCase.apiClientNil {
+			testSettings = nil
+		}
+
+		builderResult, err := Pull(testSettings, testCase.testName, testCase.testNamespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testResourceQuota.Name, builderResult.Object.Name)
+			assert.Equal(t, testResourceQuota.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestResourceQuotaWithQuotaSpec(t *testing.T) {
```

Comment:
function needs renamed for this resource

Reply:
Done.